shared/rest/src/main/java/io/pravega/shared/rest/impl/RESTServerConfigImpl.java

```diff
@@ -34,20 +35,26 @@
     private final boolean authorizationEnabled;
     private final String userPasswordFile;
     private final boolean tlsEnabled;
+    private final String tlsProtocolVersion;
     private final String keyFilePath;
     private final String keyFilePasswordPath;
 
     @Builder
     RESTServerConfigImpl(final String host, final int port, boolean authorizationEnabled, String userPasswordFile,
-                         boolean tlsEnabled, String keyFilePath, String keyFilePasswordPath) {
+                         boolean tlsEnabled, String tlsProtocolVersion, String keyFilePath, String keyFilePasswordPath) {
         Exceptions.checkNotNullOrEmpty(host, "host");
         Exceptions.checkArgument(port > 0, "port", "Should be positive integer");
         Exceptions.checkArgument(!tlsEnabled || !Strings.isNullOrEmpty(keyFilePath),
                 "TLS", "KeyFilePath should not be empty when TLS is enabled. ");
+        if (tlsEnabled && tlsProtocolVersion != null) {
+            Exceptions.checkArgument(TLSProtocolVersion.parse(tlsProtocolVersion) != null,
```

Comment:
Just parse it once and hold onto the parsed class. Do not just parse it just for the sake of validation.

Reply:
Fixed. Parsing is done only in `serviceConfig `and `Config`. Removed the parsing from the components.